plugins/scaffolder/src/extensions/types.ts

```diff
@@ -13,10 +13,15 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { ApiHolder } from '@backstage/core-plugin-api';
 import { FieldValidation, FieldProps } from '@rjsf/core';
 
 export type FieldExtensionOptions<T = any> = {
   name: string;
   component: (props: FieldProps<T>) => JSX.Element | null;
-  validation?: (data: T, field: FieldValidation) => void;
+  validation?: (
+    data: T,
+    field: FieldValidation,
+    context: { apiHolder: ApiHolder },
```

Comment:
This here too? `context?:` otherwise it's breaking.

Reply:
Or actually maybe we can just overload this, maybe that's the right way.

Reply:
Not sure if it's possible in typescript though, would have to check

Reply:
Yeah I think that's the problem, that we won't be able to 🤔 Let me know if you find something

Reply:
OK decided to go with a union type for this, so that the end user can choose whether they want to make use of the parameter or not. This also demoted the changeset to patch level